core/vm/evm.go

```diff
@@ -86,7 +86,7 @@ func (evm *EVM) precompile(addr common.Address) (precompile.StatefulPrecompiledC
 	}
 
 	// Enable the optional stateful precompiles
-	if evm.chainRules.IsAllowListEnabled {
+	if evm.chainRules.IsPrecompileEnabled(precompile.AllowListName) {
```

Comment:
We should be able to return a map of the addresses to the stateful precompile singletons to clean this up further and remove the need for adding `Name()` to the interface.

Reply:
I changed it, hope this is what you meant.